test/accesskeys.test.js

```diff
@@ -0,0 +1,546 @@
+        t.end();
+    })
+});
+
+test('validate accesskey (temporary key - cannot modify accesskeysecret)',
+     function (t) {
+    const changedTempEntry = cloneObj(tempKeyEntry);
+    changedTempEntry.dn = {
+        rdns:[
+            {accesskeyid: 'MSTS4F2A1B3C9E7D8A6F'},
+            {uuid: '930896af-bf8c-48d4-885c-6573a94b1853'},
+            {ou: 'users'},
+            {o: 'smartdc'}
+        ],
+        rdnSpaced:true,
+        length:4
+    };
+    changedTempEntry.attributes._parent = [
+        'uuid=930896af-bf8c-48d4-885c-6573a94b1853, ou=users, o=smartdc'
+    ];
+
+    const changes = [
+        {_modification: {type: 'accesskeysecret'}}
+    ];
+    accesskey.validate(changedTempEntry, config, changes, function (err) {
+        t.equal(err.name, 'ConstraintViolationError');
+        t.equal(err.message, readonlyAttrMsg);
+        t.end();
+    })
+});
+
+test('validate accesskey (temporary key - cannot modify created)',
+     function (t) {
+    const changedTempEntry = cloneObj(tempKeyEntry);
+    changedTempEntry.dn = {
+        rdns:[
+            {accesskeyid: 'MSTS4F2A1B3C9E7D8A6F'},
+            {uuid: '930896af-bf8c-48d4-885c-6573a94b1853'},
+            {ou: 'users'},
+            {o: 'smartdc'}
+        ],
+        rdnSpaced:true,
+        length:4
+    };
+    changedTempEntry.attributes._parent = [
+        'uuid=930896af-bf8c-48d4-885c-6573a94b1853, ou=users, o=smartdc'
+    ];
+
+    const changes = [
+        {_modification: {type: 'created'}}
+    ];
+    accesskey.validate(changedTempEntry, config, changes, function (err) {
+        t.equal(err.name, 'ConstraintViolationError');
+        t.equal(err.message, readonlyAttrMsg);
+        t.end();
+    })
+});
+
+test('validate accesskey (temporary key - can modify mutable fields together)',
+     function (t) {
+    const changedTempEntry = cloneObj(tempKeyEntry);
+    changedTempEntry.dn = {
+        rdns:[
+            {accesskeyid: 'MSTS4F2A1B3C9E7D8A6F'},
+            {uuid: '930896af-bf8c-48d4-885c-6573a94b1853'},
+            {ou: 'users'},
+            {o: 'smartdc'}
+        ],
+        rdnSpaced:true,
+        length:4
+    };
+    changedTempEntry.attributes.status = ['Inactive'];
+    changedTempEntry.attributes.description = ['Updated description'];
+    changedTempEntry.attributes.expiration =
+             [new Date(Date.now() + 48 * 60 * 60 * 1000).toISOString()];
+    changedTempEntry.attributes.updated = ['1763670000000'];
+    changedTempEntry.attributes._parent = [
+        'uuid=930896af-bf8c-48d4-885c-6573a94b1853, ou=users, o=smartdc'
+    ];
+
+    const changes = [
+        {_modification: {type: 'status'}},
+        {_modification: {type: 'description'}},
+        {_modification: {type: 'expiration'}},
+        {_modification: {type: 'updated'}}
+    ];
+    accesskey.validate(changedTempEntry, config, changes, function (err) {
+        t.notOk(err, 'should allow modifying mutable fields');
+        t.end();
+    })
+});
```

Comment:
I think you need to add a test where you:

- Create a temporary key with a SHORT lifetime (60s)

- Confirm it gets added AND you can use it immediately.

- Wait >= SHORT lifetime (i.e. >= 60s)

- Try using or seeking it again; it should STOP WORKING.